src/components/atoms/Accordion/Accordion.tsx

```diff
@@ -1,96 +1,144 @@
-import React, { useState, useEffect, useRef } from 'react';
+import React, { useState, useEffect, useRef, useCallback } from 'react';
 
-import { KeyboardArrowDown, KeyboardArrowUp } from '@groww-tech/icon-store/mi';
+import { KeyboardArrowDown } from '@groww-tech/icon-store/mi';
 
 import AnimateHeight from '../AnimateHeight';
 
 import './accordion.css';
 
 
 const Accordion = (props: Props) => {
+  const {
+    onMountOpen,
+    onToggleCallback,
+    title,
+    children,
+    parentClass,
+    headerClass,
+    iconClass,
+    titleClass,
+    showRightIcon,
+    useAnimateHeight,
+    maxHeight
+  } = props;
+
+  const [isOpen, toggleAccordion] = useState(onMountOpen);
 
+  const childRef = useRef<HTMLDivElement>(null);
   const isFirstRender = useRef(true);
-  const [ isOpen, setAccordionToggle ] = useState(props.onMountOpen);
+
 
   useEffect(() => {
     if (isFirstRender.current) {
```

Comment:
this check doesn't need isOpen as a dependency. We can create two separate useEffects for these.

Reply:
Both the above checks are interdependent.
The check isFirstRender is used so as to not call the callback during the first render but later everytime isOpen variable changes. So both need to be in the same useEffect